Notes/NewNoteViewController.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  NewNoteViewController.swift
+//  Notes
+//
+//  Created by Kathy Nguyen on 2/22/21.
+//
+
+import UIKit
+
+class NewNoteViewController: UIViewController {
+    @IBOutlet var titleField: UITextField!
+    @IBOutlet var noteField: UITextView!
+    
+    // used to get data from this controller
+    public var completion: ((String, String) -> Void)?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // autofocuses here first and pull up a keyboard
+        titleField.becomeFirstResponder()
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Save", style: .done, target: self, action: #selector(didTapSave))
+    }
+    
+    @objc func didTapSave() {
+        // if there is text and text isn't empty
+        if let text = titleField.text, !text.isEmpty, !noteField.text.isEmpty {
+            completion?(text, noteField.text)
```

Comment:
Why did you chose the callback option rather than delegation?

Reply:
I wasn't familiar with delegation. I had trouble setting the public variable as typealias and later calling it in ViewController. I rewrote this as delegation. Delegation was easier to pass data back up to ViewController